### PR TITLE
Enable and add missing doc tests against sandbox API

### LINKED ISSUE #37: Update urls to new documentation
Safaricom recently updated their docs and old urls to the docs used in our doc strings no longer work.

### LINKED ISSUE #82: Enable doc tests and run them against sandbox environment

### PR DESCRIPTION
**Pull request by c12i**
Closes #82
Closes #37 

# Changes
- Enable and move doc tests to `/docs/**/*.md`
- Enable doc tests in the `README.md` file
- Remove examples from `README.md` and replace with table that links to the new docs in `docs/client`, [see new README.md](https://github.com/c12i/mpesa-rust/blob/add-doc-tests-move-docs-to-md/README.md)
- Reference new Safaricom docs

# Fixes
- Fix typo in `c2b_register` response that did not account for a typo in the actual API response leading to deserialization errors
- Add `Reversal = 11` `IndentifierType` that is required for the `transaction_reversal` api, the original default `ShortCode = 4` always produced a `400` error
- Fix `400` response without a `body` when sending a `dynamic_qr` request where the `amount` is a `f64` type. The API only expects this field to be a number (or its string representation) but not a decimal. cc @itsyaasir 

# TODO

- [x] Try `cargo nextest` to handle flaky tests -> doc tests not supported, therefore we won't benefit from the retry capabilities `nextest` provides
- [x] Enable `bill_manager` API doc tests, tests added but not ran because of inconsistency between our sdk and current api implementation

# Findings
- https://github.com/c12i/mpesa-rust/pull/90#issuecomment-1818705150

### CODE REVIEW

**Comment by collinsmuriuki** [comment]
Flaky tests:

- c2b register

File: src/services/c2b_register.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ struct C2bRegisterPayload<'mpesa> {
 
 #[derive(Debug, Deserialize, Clone)]
 pub struct C2bRegisterResponse {
-    #[serde(rename(deserialize = "OriginatorConverstionID"))]
+    #[serde(rename(deserialize = "OriginatorCoversationID"))]
```
Comment (collinsmuriuki) [inline src/services/c2b_register.rs:23]: this unique typo only appears on this API

**Comment by collinsmuriuki** [comment]
The C2b simulate API seems to not exist in the docs anymore

**Comment by c12i** [comment]
Hey @crispinkoech I noticed that whenever I call the `bill_manager` APIs in sandbox, I get a `401` invalid access token error. Is this something you had encountered when you tested this out?

**Comment by crispinkoech** [comment]
> Hey @crispinkoech I noticed that whenever I call the bill_manager APIs in sandbox, I get a 401 invalid access token error. Is this something you had encountered when you tested this out?

I don't recall, it's been a while 😅

**Comment by c12i** [comment]
> > Hey @crispinkoech I noticed that whenever I call the bill_manager APIs in sandbox, I get a 401 invalid access token error. Is this something you had encountered when you tested this out?
> 
> I don't recall, it's been a while 😅

No worries, I figured out the cause, my app did not have access to bill manager.

ION, the bill manager api is due for a refactor, I noticed that there are some new changes to the APIs that are break the current implementation, will create an issue for this.

**Comment by crispinkoech** [comment]
No worries. You can assign that to me (as well as the other issue I created, seems I don't have access for that)

File: docs/client/c2b_register.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+Register URL API works hand in hand with Customer to Business (C2B) APIs and allows receiving payment notifications to your paybill. This API enables you to register the callback URLs via which you shall receive notifications for payments to your pay bill/till number.
+
+There are two URLs required for Register URL API: Validation URL and Confirmation URL.
+
+Returns a `C2bRegisterBuilder`
+
+See more from the Safaricom API docs [here](https://developer.safaricom.co.ke/APIs/CustomerToBusinessRegisterURL)
+
+# Example
+```rust,no_run
```
[the remark this answers is not in the document]
Comment (itsyaasir) [reply docs/client/c2b_register.md:10]: It is flaky, we can just ignore it for now.

**Review by itsyaasir — APPROVED** [review]

**Comment by itsyaasir** [comment]
@c12i Quick suggestion; The docs are looking good, should we add more information to the module about the service itself, maybe example response or example request ? 

We can include the docs in the module itself, something like `#![doc = include_str!("../../docs/client/dynamic_qr.md")]
`

**Pull request merged**